### DISCUSSION TITLE
Pre-configuring plugins (DALLE) with API keys

### DISCUSSION THREAD

**Discussion by FedeC137** [danny-avila/LibreChat#3338] [Q&A]
Hi everyone. Just wanted to say, love the app!. I've been playing around with it the last week.

Need help on an issue. Can i "Pre" Configure a plugin?. For example i want the plugin Dalle3 to be set with an API KEY, so that the user only needs to install it (and not provide a new one to install the plugin).

Haven't found this in the docs, but i could have missed it.

The plugin works, just want to avoid the user to input this key.

Thanks in advanced!.

**Comment by danny-avila** [comment]
Yes you can pre-configure the API key for DALL-E!

https://www.librechat.ai/docs/configuration/dotenv#dall-e

**Comment by FedeC137** [comment]
Hi Danny, first of all, huge fan, love your work!.

I tried that before posting, however when i install the DALLE 3 plugin, it requires me to set a OpenAi API Key, even if i set it in the .env file. 
Am i doing something wrong?, can i use DALLE without the Plugin ? 

Thanks in advanced.

![Screenshot 2024-07-15 193351](https://github.com/user-attachments/assets/e5d61c09-9b08-448a-8031-bc2360e6beb9)

**Comment by FedeC137** [comment]
Got it to work.
I think modified the .env file after installing the plugin. So i uninstalled it / reinstalled it, and its working now. Thanks!
